src/libs/freetype/lv_freetype.c

```diff
@@ -360,9 +360,14 @@ static FTC_FaceID lv_freetype_req_face_id(lv_freetype_context_t * ctx, const cha
     node = _lv_ll_ins_tail(ll_p);
     LV_ASSERT_MALLOC(node);
 
-    node->pathname = lv_malloc(len + 1);
+    if(LV_FS_MEMFS_LETTER == pathname[0]){
+        len = sizeof(lv_fs_path_ex_t);
+    } else {
+        len++;
+    }
+    node->pathname = lv_malloc(len);
     LV_ASSERT_MALLOC(node->pathname);
-    strcpy(node->pathname, pathname);
+    memcpy(node->pathname, pathname, len);
```

Comment:
Please use `lv_strdup`, and you can remove `lv_malloc` and `LV_ASSERT_MALLOC`
https://github.com/lvgl/lvgl/blob/master/src/stdlib/builtin/lv_string_builtin.c#L209

See example usage: https://github.com/lvgl/lvgl/blob/master/src/draw/lv_image_decoder.c#L241

Reply:
Hi @PGNetHun ,

Actually this can't be used here because when you use memfs the path is actually a pointer to a `lv_fs_path_ex_t` object. The first field of this structure is a null terminated string.
When cast to `char *` subsequent use of strlen will only copy the first field of the structure thus leading to the loss of the relevant metadata for proper use of memfs. 
I hope that clears out why I used this hack on `len` and `malloc`.

Reply:
Yes, that makes sense.